### ISSUE TITLE
pl.DataFrame.explode() gives an empty name name now for the exploded column.

### ISSUE THREAD

**Issue by ghuls** [pola-rs/polars#1391]
#### Are you using Python or Rust?

 **Python**.

#### What version of polars are you using?

0.9.8

#### What operating system are you using polars on?

CentOS7

#### Describe your bug.

Before polars =<0.9.3, `pl.DataFrame.explode()` would keep the exploded column name.
from 0.9.4 and onwards it is set to an empty name.

Is this the intended behaviour? How can I rename a column with no name?

```python
In [7]: df = pl.DataFrame({
   ...:     "letters": ["c", "c", "a", "c", "a", "b"],
   ...:     "nrs": [[1, 2], [1, 3], [4, 3], [5, 5, 5], [6], [2, 1, 2]]
   ...: })

In [8]: df.explode("nrs")
Out[8]: 
shape: (13, 2)
┌─────────┬─────┐
│ letters ┆     │
│ ---     ┆ --- │
│ str     ┆ i64 │
╞═════════╪═════╡
│ "c"     ┆ 1   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "c"     ┆ 2   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "c"     ┆ 1   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "c"     ┆ 3   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ ...     ┆ ... │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "c"     ┆ 5   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "a"     ┆ 6   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "b"     ┆ 2   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "b"     ┆ 1   │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌┤
│ "b"     ┆ 2   │
└─────────┴─────┘
```

The `pl.DataFrame.explode()` example in the polars code has the output with the column name.

**Comment by ritchie46** [comment]
> Is this the intended behaviour?

This is a bug. Will fix that.

**Comment by ritchie46** [comment]
A bit of context. The explode is almost a no-op if we can proof that a list array has got no empty values: `[[1, 2], [1], [2, 3]]` vs `[[1, 2]. [], [2, 3]]`. In the first case, we can just take the value array as a new series. In the latter we must do a lot more compute and copying. 

During creation of a List array we track statistics of this. However, the list array in python is created in pyarrow and in that case we don't have this stats and we must do the more expensive explode. That's what triggered the change.

Fix in#1392

**Comment by ritchie46** [comment]
Fixed and released.
